src/viewer.js

```diff
@@ -742,8 +742,9 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
             return;
         }
 
-        if (this.source.freeupCanvasMemory) {
-            this.source.freeupCanvasMemory();
+        //TODO: implement destroy and _freeupCanvasMemory method for all child classes of TileSource, then remove if statement wrap below.
```

Comment:
I think it's fine for destroy to be optional, so I don't think we need this TODO.